tests/python_frontend/power_operator_test.py

```diff
@@ -35,8 +37,35 @@ def test_pow_neg_exp():
     pow_neg_exp(a=res)
     assert (res[0] == 0.01)
 
+in_types = [dace.float32, dace.float64, dace.int8, dace.int16, dace.int32, dace.int64]
+
+@pytest.mark.parametrize("a_type", in_types)
+@pytest.mark.parametrize("b_type", in_types)
+def test_pow_types(a_type, b_type):
+    @dace.program
+    def pow_types(A: a_type[1], B: b_type[1], R: dace.float64[1]):
+        with dace.tasklet(dace.Language.Python):
+            a << A[0]
+            b << B[0]
+            r >> R[0]
+            """r = a ** b"""
```

Comment:
why is this a string and not just `r = a ** b`?

Reply:
If we don't use the string, the generated CPP uses pointers directly instead of reading out the array. Perhaps an optimization bug that needs to be investigated. Alternatively we could add wrappers for pointer types.

![image](https://github.com/spcl/dace/assets/71826159/9c61bb56-35da-49d6-95df-6ca9dc050f93)

Reply:
Oof, good catch. I quickly checked the SDFG (without string/as code) and it contains a tasklet with `r = (None ** None)`. I guess then somehow `__g_a` and `__g_b` get introduced.

I also tried `R[0] = A[0] ** B[0]` (without the explicit tasklet), but this leads to a tasklet with code `__out = (dace.float64(__tmp0) ** dace.float64(__tmp1))`. The casts completely defeat the purpose of this PR.

It seems like using the string notation is currently the only reliable way to properly test the C++ overloads.

My suggestion would be to separate these issues from this PR. I can open GH issues with tests for those. So that we can merge this PR, since it's unrelated to these issues.

Reply:
First issue: https://github.com/spcl/dace/issues/1588
I looked into the other one as well, but I don't really understand the issue yet. I saw some interesting behavior with different types...

Reply:
I changed it based on the feedback from https://github.com/spcl/dace/issues/1588